### DISCUSSION TITLE
Custom Command with URScipt

### DISCUSSION THREAD

**Discussion by LoyWeiWin** [visose/Robots#226] [Q&A] [answered]
Hey yall,

I;m currently trying to use _robots_ to communicate with a Robotiq gripper 2F-85. Robotiq has provided a collection of custom urscrip to allow users to communicate with their grippers. However, when I'm trying to use the custom command in robots and yield no result. I just wondering if the custom command only support the default urscript provided by the universal robot? Kindly reply.

Thanks =)

Kind regards
Loy

**Comment by visose** [comment, marked as answer]
They should work if the controller has been configured to work with them. Try first using them in a normal program created using the pendant. If that works, check the generated code from Robots for anything unusual, like wrong indentation.

**Reply by jrojasr1986** [reply]
Hello,

I tried something similar. Not sure if we can call directly the functions from robotiq for a Hand-E gripper. I tried from the teach pendant and the function works. But when I am sending it through the scripts not working. Any thoughts about how to fix my code line?

![Capture](https://github.com/visose/Robots/assets/130080096/cdfa0392-1400-4792-97a5-83272bc86a6d)
![WhatsApp Image 2024-03-27 at 3 53 38 PM](https://github.com/visose/Robots/assets/130080096/75667e1f-c7dd-4712-ad30-72acd05da005)

**Reply by jrojasr1986** [reply]
@visose you are right I was able to call the commands from Robotiq.

**Comment by LoyWeiWin** [comment]
Since the mechanical coupling that connects the Robotiq gripper to the UR10 tool flange is missing, the custom command method doesn’t work. Instead, we ended up communicating with the control box via RS485
